pkg/snapstore/ocs_s3_snapstore.go

```diff
@@ -28,32 +41,166 @@ const (
 	ocsSecretAccessKey    string = "OCS_SECRET_ACCESS_KEY"
 )
 
+type ocsAuthOptions struct {
+	Endpoint           string `json:"endpoint"`
+	Region             string `json:"region"`
+	DisableSSL         bool   `json:"disableSSL"`
+	InsecureSkipVerify bool   `json:"insecureSkipVerify"`
+	AccessKeyID        string `json:"accessKeyID"`
+	SecretAccessKey    string `json:"secretAccessKey"`
+}
+
 // NewOCSSnapStore creates a new S3SnapStore from shared configuration with the specified bucket.
 func NewOCSSnapStore(config *brtypes.SnapstoreConfig) (*S3SnapStore, error) {
-	ao, err := ocsAuthOptionsFromEnv()
+	credentials, err := getOCSAuthOptions(getEnvPrefixString(config.IsSource))
+	if err != nil {
+		return nil, err
+	}
+
+	return newGenericS3FromAuthOpt(config.Container, config.Prefix, config.TempDir, config.MaxParallelChunkUploads, ocsAuthOptionsToGenericS3(*credentials))
+}
+
+func getOCSAuthOptions(prefix string) (*ocsAuthOptions, error) {
+	if _, isSet := os.LookupEnv(prefix + ocsCredentialFileJSONFile); isSet {
+		if filename := os.Getenv(prefix + ocsCredentialFileJSONFile); filename != "" {
+			ao, err := readOCSCredentialsJSON(filename)
+			if err != nil {
+				return nil, fmt.Errorf("error getting credentials using %v file", filename)
+			}
+			return ao, nil
+		}
+	}
+
+	if _, isSet := os.LookupEnv(prefix + ocsCredentialFile); isSet {
+		if dir := os.Getenv(prefix + ocsCredentialFile); dir != "" {
+			ao, err := readOCSCredentialFromDir(dir)
+			if err != nil {
+				return nil, fmt.Errorf("error getting credentials from %v directory", dir)
+			}
+			return ao, nil
+		}
+	}
+
+	// TODO: passing credentials through environment variable will be deprecated by "backup-restore v0.18.0"
+	if _, isSet := os.LookupEnv(prefix + ocsEndpoint); isSet {
+		ao, err := readOcsCredentialFromEnv()
+		if err != nil {
+			return nil, fmt.Errorf("error getting credentials from env: %s", err.Error())
+		}
+		return ao, nil
+	}
+
+	return nil, fmt.Errorf("unable to get credentials")
+}
+
+func readOCSCredentialFromDir(dirname string) (*ocsAuthOptions, error) {
+	ao := ocsAuthOptions{}
+
+	files, err := ioutil.ReadDir(dirname)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, file := range files {
+		switch file.Name() {
+		case "endpoint":
+			{
+				data, err := ioutil.ReadFile(dirname + "/endpoint")
+				if err != nil {
+					return nil, err
+				}
+				ao.Endpoint = string(data)
+			}
+		case "accessKeyID":
+			{
+				data, err := ioutil.ReadFile(dirname + "/accessKeyID")
```

Comment:
```suggestion
				data, err := os.ReadFile(dirname + "/accessKeyID")
```
same as above and for all places